js_dependencies/Bonito.js

```diff
@@ -65,6 +65,45 @@ function fetch_binary(url) {
     });
 }
 
+// from: https://www.geeksforgeeks.org/javascript-throttling/
+function throttle_function(func, delay) {
+    // Previously called time of the function
+    let prev = 0;
+    // ID of queued future update
+    let future_id = undefined;
+    function inner_throttle(...args) {
+        // Current called time of the function
+        const now = new Date().getTime();
+
+        // If we had a queued run, clear it now, we're
+        // either going to execute now, or queue a new run.
+        if (future_id !== undefined) {
+            clearTimeout(future_id);
+            future_id = undefined;
+        }
+
+        // If difference is greater than delay call
+        // the function again.
+        if (now - prev > delay) {
+            prev = now;
+            // "..." is the spread operator here
+            // returning the function with the
+            // array of arguments
+            return func(...args);
+        } else {
+            // Otherwise, we want to queue this function call
+            // to occur at some later later time, so that it
+            // does not get lost; we'll schedule it so that it
+            // fires just a bit after our choke ends.
+            future_id = setTimeout(
+                () => inner_throttle(...args),
+                now - prev + 1
+            );
```

Comment:
I just noticed that our implementation tries to not busy wait like the original implementation via setTimeout.
But I'm a bit confused by the number, shouldn't this be something like `Math.max(1, delay - (now - prev))`?

Reply:
indeed!  but i don't think the max with 1 is necessary because the other branch in the if/else block will execute if the second arg is negative.  i'll push a commit...